src/strings_with_arrows.py

```diff
@@ -0,0 +1,26 @@
+def string_with_arrows(text, pos_start, pos_end):

```

Comment:
What is `string_with_arrows`? I think this is the most puzzling bit of all the code. Documentation here would greatly help.